### DISCUSSION TITLE
BT shows up on Android phone but not on 7th gen ipad

### DISCUSSION THREAD

**Discussion by Guustaaf** [micropython/micropython#11794] [RP2040 / Pico] [answered]
I would love to have it show up on iPads also.

Thanks

`import bluetooth
import aioble
import uasyncio as asyncio
from micropython import const

_ENV_SENSE_UUID = bluetooth.UUID(0x181A)
_ENV_SENSE_TEMP_UUID = bluetooth.UUID(0x2A6E)
_GENERIC_THERMOMETER = const(768)
_ADV_INTERVAL_MS = const(10000)

temp_service = aioble.Service(_ENV_SENSE_UUID)
temp_char = aioble.Characteristic(temp_service, _ENV_SENSE_TEMP_UUID, read=True, notify=True)

aioble.register_services(temp_service)

async def main():
    while True:
        print("Starting to advertise...")
        try:
            connection = await aioble.advertise(
                    _ADV_INTERVAL_MS,
                    name="Elixonic",
                    services=[_ENV_SENSE_UUID],
                    appearance=_GENERIC_THERMOMETER,
                    manufacturer=(0x0bcd, b"1234"),
                )
            print("Connection from", connection.device)
        except Exception as e:
            print("Error: ", e)
            await asyncio.sleep(1)

# Run the main function until it is complete
print("Starting the main function...")
asyncio.run(main())
`

**Comment by Guustaaf** [comment]
It looks like iPads only work with regular bluetooth, not BLE.

Can the Pico do regular BT?

**Reply by jimmo** [reply]
iPads can definitely do BLE. (Possibly unless it's a _really_ old iPad?)

You should be able to use the nRF Connect to see the Pico W.

**Comment by fdufnews** [comment]
On  [Raspberry Pi internet site](https://www.raspberrypi.com/news/new-functionality-bluetooth-for-pico-w/) Eben said

"Pico W’s wireless functionality is provided by the Infineon [CYW43439](https://www.infineon.com/cms/en/product/wireless-connectivity/airoc-wi-fi-plus-bluetooth-combos/wi-fi-4-802.11n/cyw43439/) device, which contains a 2.4 GHz radio providing both 802.11n Wi-Fi and Bluetooth 5.2, supporting Bluetooth Classic and Bluetooth Low Energy (BLE) functionality. At launch, our firmware and host-side software only enabled Wi-Fi, not Bluetooth.
Now, with the release of version [1.5.1](https://github.com/raspberrypi/pico-sdk/releases/tag/1.5.1) of the C SDK, and the [latest MicroPython build](https://datasheets.raspberrypi.com/soft/micropython-firmware-pico-w-130623.uf2), your existing Pico W has gained the ability to communicate with Bluetooth and Bluetooth LE devices. Specifically, we support Bluetooth Classic — with the temporary exception of ACL/SCO — along with both the BLE Central and Peripheral roles. Things are also configurable so you can enable Bluetooth Classic and BLE either individually, or have both of them available at the same time. Full details of [supported Bluetooth protocols and profiles](https://github.com/bluekitchen/btstack#supported-protocols-and-profiles) are available on GitHub."

**Reply by jimmo** [reply, marked as answer]
This quote is a bit misleading. The C SDK supports BLE and Classic. _MicroPython_ only supports BLE.
